zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/archiver/OpenSearchArchiverRepository.java

```diff
@@ -195,6 +190,43 @@ public CompletableFuture<Void> reindexDocuments(
         .thenApplyAsync(ignored -> null, executor);
   }
 
+  private CompletableFuture<List<String>> fetchIndexMatchingIndexes(final String indexWildCard)
+      throws IOException {
+    return client
+        .cat()
+        .indices()
+        .thenApply(
+            response ->
+                response.valueBody().stream()
+                    .map(IndicesRecord::index)
+                    .filter(index -> index.matches(indexWildCard))
+                    .toList());
+  }
+
+  private CompletableFuture<Void> applyPolicyToIndex(final String index) {
+    final AddPolicyRequestBody value = new AddPolicyRequestBody(retention.getPolicyName());
+    final var request = Requests.builder().method("POST").endpoint("_plugins/_ism/add/" + index);
+    return sendRequestAsync(
+            () ->
+                genericClient.executeAsync(
+                    request.json(value, genericClient._transport().jsonpMapper()).build()))
+        .thenComposeAsync(
```

Comment:
:x: Specify the executor as well to ensure we don't execute the callback after shutdown.